### PR TITLE
Initialize AshGraphQL and port tenantInfo query

### PR DESCRIPTION
**Pull request by rbino**
Also revamp auth and context generation, leveraging Ash utilities

### CODE REVIEW

File: backend/test/support/conn_case.ex [hunk 1]
```diff
@@ -68,18 +68,8 @@ defmodule EdgehogWeb.ConnCase do
...
+    # Create a tenant fixture
     tenant = Edgehog.TenantsFixtures.tenant_fixture(public_key: tenant_public_key)
-    _ = Edgehog.Repo.put_tenant_id(tenant.tenant_id)
-
-    conn =
-      if tags[:unauthenticated] do
-        # Just return the conn
-        conn
-      else
-        authenticate_connection(conn, tenant_private_key)
```
Comment (davidebriani) [inline backend/test/support/conn_case.ex:81]: You can remove the `authenticate_connection` function if it's not used anymore.

File: backend/test/edgehog_web/auth_test.exs [hunk 2]
```diff
@@ -19,11 +19,11 @@
 #
 
 defmodule EdgehogWeb.AuthTest do
-  use EdgehogWeb.ConnCase, async: true
+  # This can't be async: true since it modifies the Application env
```
Comment (davidebriani) [inline backend/test/edgehog_web/auth_test.exs:22]: I'm not sure what this refers to, the `ConnCase` doesn't seem to touch the `Application` env, at least directly.

**Review by davidebriani — APPROVED** [review]

File: backend/test/support/conn_case.ex (hunk 1 above)
Comment (rbino) [reply backend/test/support/conn_case.ex:81]: It's actually used in `auth_test` (it gets imported there)

**Review by szakhlypa — CHANGES_REQUESTED** [review]

File: backend/lib/edgehog/tenants/tenant/tenant.ex [hunk 3]
```diff
@@ -21,14 +21,34 @@
...
+
+    # We don't care about filtering here
+    derive_filter? false
+    # :tenant_id, the primary key, already gets exposed as ID, so we hide it here
+    # to avoid showing it twice. We also hide the public key to be consistent with
+    # the old API
+    hide_fields [:tenant_id, :public_key]
+
+    queries do
+      read_one :tenant_info, :current_tenant
```
[the remark this answers is not in the document]
Comment (rbino) [reply backend/lib/edgehog/tenants/tenant/tenant.ex:48]: Fixed both

**Review by szakhlypa — APPROVED** [review]

**Pull request merged**